### DISCUSSION TITLE
SAML still broken for me in 2.5.281

### DISCUSSION THREAD

**Discussion by Ma27** [requarks/wiki#5289] [Error / Bug Report] [answered]
With wiki-js at 2.5.281 I get the following error when trying to do a SAML auth:
```
Oops, something went wrong...
this.options.authnContext.forEach is not a function
```

The configuration used to work fine with 2.5.279.

I have set the following settings:
* Entry point: https://ldap-hostname/saml/saml2/idp/SSOService.php
* Issuer: passport-saml-wikijs
* Certificate: self-signed cert
* Name identifier format: urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress
* Auth context: urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport
* Provider name: wiki.js
* Request binding: HTTP-Redirect
* Unique ID Field Mapping: samlLoginName
* Email Field Mapping: emailAddress
* Display name Field Mapping: samlLoginName
* Avatar picture field mapping: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/picture

**Comment by Ma27** [comment, marked as answer]
Perhaps I found a potential fix (see also https://github.com/node-saml/passport-saml/pull/615), `authnContext` must be an array now.

Currently deploying (though this may take a while since I'm on a German train ^^), if it works I'd file a PR later.

**Reply by washcroft** [reply]
@Ma27 Did the PR work for you? Since getting over the `this.options.authnContext.forEach is not a function error`, I now have the error message `Invalid authentication provider.` instead.

**Reply by Ma27** [reply]
@washcroft it's working perfectly fine for me on 2.5.282 :)
